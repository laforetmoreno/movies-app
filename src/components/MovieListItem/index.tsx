import React from "react";

import TagList from "../TagList";

import { Movie } from "../../types";

import style from "./style.module.scss";

interface Props {
  movie?: Movie;
}

const MovieListItem = ({ movie }: Props) => (
  <li>
    <a
      className={style.wrapper}
      rel="noopener noreferrer"
      target="_blank"
      href={movie.trailers[0].url ? movie.trailers[0].url : movie.siteURL}
    >
      <img className={style.image} alt={movie?.title} src={movie?.images[0]?.url} />
      <span className={style?.title}>{movie?.title}</span>
      <TagList tags={movie.tags} />
    </a>
  </li>
);

export default MovieListItem;
