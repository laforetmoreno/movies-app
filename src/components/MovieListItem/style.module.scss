@import "../../styles/variables";

.wrapper {
  color: $white;
  display: flex;
  flex-direction: column;
  text-decoration: none;
  margin-bottom: 30px;
  position: relative;

  &:hover {
    transition: transform 0.2s ease-in-out, -webkit-transform 0.2s ease-in-out;
    transform: translateY(-5px);
  }

  .title {
    margin-top: 10px;
  }

  .image {
    @media (max-width: 767px) {
      width: 100%;
    }

    @media (min-width: 768px) {
      width: 300px;
    }

    @media (min-width: 1200px) {
      width: 200px;
    }
  }
}
