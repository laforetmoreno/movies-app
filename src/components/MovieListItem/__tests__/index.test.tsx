import React from "react";
import Link from "../Link.react";
import renderer from "react-test-renderer";

import MovieListItem from "../index";

it("renders correctly", () => {
  const defaultProps = {
    movie: {
      trailers: [""],
      images: [""],
      tags: [""]
    }
  };
  const tree = renderer.create(<MovieListItem {...defaultProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});
