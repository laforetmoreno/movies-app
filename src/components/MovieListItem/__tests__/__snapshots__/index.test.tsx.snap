// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li>
  <a
    className="wrapper"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      className="image"
    />
    <span
      className="title"
    />
    <ul
      className="tags"
    >
      <li
        className="tag undefined"
      >
        
      </li>
    </ul>
  </a>
</li>
`;
